### PR TITLE
Migrate to OpenSearch

### PR DESCRIPTION
**Pull request by ewdurbin**
We migrated to an OpenSearch cluster on AWS, so update local dev to match.

### CODE REVIEW

File: dev/compose/opensearch/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM opensearchproject/opensearch:2.12.0
+
+RUN opensearch-plugin remove opensearch-skills --purge
+RUN opensearch-plugin remove opensearch-ml --purge
+RUN opensearch-plugin remove opensearch-neural-search --purge
+RUN opensearch-plugin remove opensearch-performance-analyzer --purge
+RUN opensearch-plugin remove opensearch-security-analytics --purge
```
Comment (ewdurbin) [inline dev/compose/opensearch/Dockerfile:7]: This culls a bunch of pointless plugins (for local dev) that just slow startup times down a ton.

File: docker-compose.yml [hunk 2]
```diff
@@ -44,11 +44,20 @@ services:
...
+        test: ["CMD-SHELL", "curl -u admin:gqYeDIzbEwTTYmB7 --silent --fail http://localhost:9200/_cluster/health || exit 1"]
+        interval: 1s
+        start_period: 10s
     environment:
-      - xpack.security.enabled=false
       - discovery.type=single-node
+      - OPENSEARCH_INITIAL_ADMIN_PASSWORD=gqYeDIzbEwTTYmB7
+      - DISABLE_INSTALL_DEMO_CONFIG=true
+      - DISABLE_SECURITY_PLUGIN=true
+      - DISABLE_PERFORMANCE_ANALYZER_AGENT_CLI=true
```
Comment (ewdurbin) [inline docker-compose.yml:60]: This culls a bunch of pointless plugins (for local dev) that just slow startup times down a ton.

**Comment by ewdurbin** [comment]
TBD on what happens with client upgrade...

**Comment by ewdurbin** [comment]
client update looking good, I plan to include it in this PR

**Comment by ewdurbin** [comment]
closes #14671

**Review by dstufft — APPROVED** [review]

**Pull request merged**